### DISCUSSION TITLE
Prettier: Fix Sorting Problems

### DISCUSSION THREAD

**Discussion by rayadaschn** [chengpeiquan/bassist#22] [Ideas]
For example, sorting of attributes, sorting of unocss class.

Problem recurrence: Since prettier is used, prettier detects that the sorting problem exposed by eslint cannot be fixed if the fix is left to prettier.

Refer: [unocss Issue](https://github.com/unocss/unocss/issues/1825)

**Comment by chengpeiquan** [comment]
I'll pay attention to it later, but I often turn off this feature because in `.tsx` a more friendly writing process can be achieved with the help of tools such as [classnames](https://github.com/JedWatson/classnames) :

```tsx
const cls = classNames(
  'flex flex-shrink-0 justify-center items-center', // Flex Layout Group
  'w-full h-50', // Size Group
  'box-border px-6', // Margin Group
  'hover:bg-brand-100', // Interactive Group
  'rounded-lg cursor-pointer', // Others Group
);
```

I have always felt that this type of grouping is more maintainable than the default sorting.

**Reply by rayadaschn** [reply]
Haha, I don't write enough react, I turned off prettier and used Eslint's own fix to fix it.

vscode ->  `settings.json`

```json
// Disable the default formatter
"prettier.enable": false,
"editor.formatOnSave": false,

// Auto fix
"editor.codeActionsOnSave": {
  "source.fixAll.eslint": true,
  "source.organizeImports": false
},
```

**Reply by chengpeiquan** [reply]
> Haha, I don't write enough react

😄 Vue is also available, it is a pure JavaScript library.
